### ISSUE TITLE
stop at Creating the hashed descriptors

### ISSUE THREAD

**Issue by zhixy** [sweeneychris/TheiaSfM#96]
When I was trying "build_construction", it sometimes stopped to do nothing at Creating the hashed descriptors, and the system monitor illustrate that nothing is been doing.
![image](https://cloud.githubusercontent.com/assets/15323271/16897579/0c69e9d4-4be8-11e6-9b4d-08bc8f6b2595.png)

**Comment by sweeneychris** [comment]
Can you give more information? Such as the output of running it with --logtostderr --v=2

**Comment by zhixy** [comment]
Hi,
Yes, here is the output.
`I0717 16:35:18.075139  4835 feature_extractor_and_matcher.cc:187] Image /home/xy/Seafile/myprivate/2011_09_26/2011_09_26_drive_0048_sync/image_00/data/jpg/0000000011.jpg did not contain an EXIF focal length.
I0717 16:35:18.077589  4837 feature_extractor_and_matcher.cc:187] Image /home/xy/Seafile/myprivate/2011_09_26/2011_09_26_drive_0048_sync/image_00/data/jpg/0000000004.jpg did not contain an EXIF focal length.
I0717 16:35:18.093170  4834 feature_extractor_and_matcher.cc:187] Image /home/xy/Seafile/myprivate/2011_09_26/2011_09_26_drive_0048_sync/image_00/data/jpg/0000000012.jpg did not contain an EXIF focal length.
I0717 16:35:18.096668  4833 feature_extractor_and_matcher.cc:187] Image /home/xy/Seafile/myprivate/2011_09_26/2011_09_26_drive_0048_sync/image_00/data/jpg/0000000016.jpg did not contain an EXIF focal length.
I0717 16:35:18.097525  4829 feature_extractor_and_matcher.cc:187] Image /home/xy/Seafile/myprivate/2011_09_26/2011_09_26_drive_0048_sync/image_00/data/jpg/0000000019.jpg did not contain an EXIF focal length.
I0717 16:35:18.099586  4843 feature_extractor_and_matcher.cc:187] Image /home/xy/Seafile/myprivate/2011_09_26/2011_09_26_drive_0048_sync/image_00/data/jpg/0000000000.jpg did not contain an EXIF focal length.
I0717 16:35:18.423506  4843 feature_extractor_and_matcher.cc:92] Successfully extracted 760 features from image /home/xy/Seafile/myprivate/2011_09_26/2011_09_26_drive_0048_sync/image_00/data/jpg/0000000000.jpg
I0717 16:35:18.431313  4843 cascade_hashing_feature_matcher.cc:116] Created the hashed descriptors for image: 0000000000.jpg
I0717 16:35:18.458823  4843 feature_extractor_and_matcher.cc:187] Image /home/xy/Seafile/myprivate/2011_09_26/2011_09_26_drive_0048_sync/image_00/data/jpg/0000000007.jpg did not contain an EXIF focal length.
I0717 16:35:18.474061  4834 feature_extractor_and_matcher.cc:92] Successfully extracted 810 features from image /home/xy/Seafile/myprivate/2011_09_26/2011_09_26_drive_0048_sync/image_00/data/jpg/0000000012.jpg
I0717 16:35:18.481003  4834 cascade_hashing_feature_matcher.cc:116] Created the hashed descriptors for image: 0000000012.jpg
I0717 16:35:18.493999  4835 feature_extractor_and_matcher.cc:92] Successfully extracted 844 features from image /home/xy/Seafile/myprivate/2011_09_26/2011_09_26_drive_0048_sync/image_00/data/jpg/0000000011.jpg
I0717 16:35:18.505022  4835 cascade_hashing_feature_matcher.cc:116] Created the hashed descriptors for image: 0000000011.jpg
I0717 16:35:18.519768  4834 feature_extractor_and_matcher.cc:187] Image /home/xy/Seafile/myprivate/2011_09_26/2011_09_26_drive_0048_sync/image_00/data/jpg/0000000017.jpg did not contain an EXIF focal length.
I0717 16:35:18.541792  4835 feature_extractor_and_matcher.cc:187] Image /home/xy/Seafile/myprivate/2011_09_26/2011_09_26_drive_0048_sync/image_00/data/jpg/0000000010.jpg did not contain an EXIF focal length.
I0717 16:35:18.566236  4837 feature_extractor_and_matcher.cc:92] Successfully extracted 844 features from image /home/xy/Seafile/myprivate/2011_09_26/2011_09_26_drive_0048_sync/image_00/data/jpg/0000000004.jpg
I0717 16:35:18.567946  4829 feature_extractor_and_matcher.cc:92] Successfully extracted 752 features from image /home/xy/Seafile/myprivate/2011_09_26/2011_09_26_drive_0048_sync/image_00/data/jpg/0000000019.jpg
I0717 16:35:18.575940  4837 cascade_hashing_feature_matcher.cc:116] Created the hashed descriptors for image: 0000000004.jpg
I0717 16:35:18.584147  4833 feature_extractor_and_matcher.cc:92] Successfully extracted 809 features from image /home/xy/Seafile/myprivate/2011_09_26/2011_09_26_drive_0048_sync/image_00/data/jpg/0000000016.jpg
I0717 16:35:18.587229  4829 cascade_hashing_feature_matcher.cc:116] Created the hashed descriptors for image: 0000000019.jpg
I0717 16:35:18.595481  4833 cascade_hashing_feature_matcher.cc:116] Created the hashed descriptors for image: 0000000016.jpg
I0717 16:35:18.609243  4837 feature_extractor_and_matcher.cc:187] Image /home/xy/Seafile/myprivate/2011_09_26/2011_09_26_drive_0048_sync/image_00/data/jpg/0000000009.jpg did not contain an EXIF focal length.
I0717 16:35:18.615346  4829 feature_extractor_and_matcher.cc:187] Image /home/xy/Seafile/myprivate/2011_09_26/2011_09_26_drive_0048_sync/image_00/data/jpg/0000000013.jpg did not contain an EXIF focal length.
I0717 16:35:18.622552  4833 feature_extractor_and_matcher.cc:187] Image /home/xy/Seafile/myprivate/2011_09_26/2011_09_26_drive_0048_sync/image_00/data/jpg/0000000014.jpg did not contain an EXIF focal length.
I0717 16:35:18.845036  4843 feature_extractor_and_matcher.cc:92] Successfully extracted 855 features from image /home/xy/Seafile/myprivate/2011_09_26/2011_09_26_drive_0048_sync/image_00/data/jpg/0000000007.jpg
I0717 16:35:18.851938  4843 cascade_hashing_feature_matcher.cc:116] Created the hashed descriptors for image: 0000000007.jpg
I0717 16:35:18.914531  4833 feature_extractor_and_matcher.cc:92] Successfully extracted 789 features from image /home/xy/Seafile/myprivate/2011_09_26/2011_09_26_drive_0048_sync/image_00/data/jpg/0000000014.jpg
I0717 16:35:18.920512  4833 cascade_hashing_feature_matcher.cc:116] Created the hashed descriptors for image: 0000000014.jpg
I0717 16:35:18.996624  4834 feature_extractor_and_matcher.cc:92] Successfully extracted 796 features from image /home/xy/Seafile/myprivate/2011_09_26/2011_09_26_drive_0048_sync/image_00/data/jpg/0000000017.jpg
I0717 16:35:19.006383  4834 cascade_hashing_feature_matcher.cc:116] Created the hashed descriptors for image: 0000000017.jpg
I0717 16:35:19.015784  4835 feature_extractor_and_matcher.cc:92] Successfully extracted 841 features from image /home/xy/Seafile/myprivate/2011_09_26/2011_09_26_drive_0048_sync/image_00/data/jpg/0000000010.jpg
I0717 16:35:19.022049  4835 cascade_hashing_feature_matcher.cc:116] Created the hashed descriptors for image: 0000000010.jpg
I0717 16:35:19.063349  4837 feature_extractor_and_matcher.cc:92] Successfully extracted 879 features from image /home/xy/Seafile/myprivate/2011_09_26/2011_09_26_drive_0048_sync/image_00/data/jpg/0000000009.jpg
I0717 16:35:19.073781  4837 cascade_hashing_feature_matcher.cc:116] Created the hashed descriptors for image: 0000000009.jpg
I0717 16:35:19.074487  4829 feature_extractor_and_matcher.cc:92] Successfully extracted 838 features from image /home/xy/Seafile/myprivate/2011_09_26/2011_09_26_drive_0048_sync/image_00/data/jpg/0000000013.jpg
I0717 16:35:19.080646  4829 cascade_hashing_feature_matcher.cc:116] Created the hashed descriptors for image: 0000000013.jpg
`

**Comment by sweeneychris** [comment]
Could you perhaps email me the dataset and the flags file that you are using to set the parameters? It's not clear from the output why this would fail, but I can diagnose it better if I have the dataset.

**Comment by zhixy** [comment]
Hi Chris,
I have sent the dataset and flag file to you. Thank you!

**Comment by zhixy** [comment]
Hi Chris,
   Interesting thing is that my Ubuntu crashed yesterday, and I reinstalled it, then I compiled Theia on the new Ubuntu and run "build_reconstruction" again, this issue disappears! Perhaps this issue results from some issues of my OS. However, I am pretty sure that I compiled Theia correctly and test successfully in old Ubuntu except for No.13 and No.47 which also failed in my new Ubuntu. I still don't know what the reason is, but anyway, it works, I am sorry to bother you and really appreciate your kind help!
